src/main/scala/where/Predicate.scala

```diff
@@ -1,10 +1,32 @@
 package com.originate.scalypher.where
 
 import com.originate.scalypher.ConstantString
+import com.originate.scalypher.path.Path
 import com.originate.scalypher.ToQuery
 
-sealed trait Predicate extends ToQuery
+case class PredicateWithProjection(
+  predicate: Predicate,
+  projection: PathProjection
+) {
+  def where(f: ObjectReference => Where): PredicateCondition =
```

Comment:
no distinction. `ObjectReference` actually just contains a `Referenceable` type (though i'll probably add a type below `Referenceable` that encompasses nodes + relationships but not paths... that will come later though)